base-lite/ub/ub_test.go

```diff
@@ -0,0 +1,329 @@
+			want:    false,
+			wantErr: false},
+		{name: "file does not exist",
+			args: args{filePath: fileDoesNotExist,
+				operation: "existence"},
+			want:    false,
+			wantErr: false},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := path(tt.args.filePath, tt.args.operation)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("path() error = %v, wantErr %v", err, tt.wantErr)
+			}
+			if got != tt.want {
+				t.Errorf("path() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_renderTemplate(t *testing.T) {
+	type args struct {
+		templateFilePath string
+	}
+	const (
+		fileExistsAndRenderable = "testResources/sampleLog4j.template"
+		fileDoesNotExist        = "testResources/RandomFileName"
+	)
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name:    "render template success",
+			args:    args{templateFilePath: fileExistsAndRenderable},
+			wantErr: false,
+		},
+		{
+			name:    "render template failure ",
+			args:    args{templateFilePath: fileDoesNotExist},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if err := renderTemplate(tt.args.templateFilePath); (err != nil) != tt.wantErr {
+				t.Errorf("renderTemplate() error = %v, wantErr %v", err, tt.wantErr)
+			}
+		})
+	}
+}
+func Test_convertKey(t *testing.T) {
+	type args struct {
+		key string
+	}
+	tests := []struct {
+		name       string
+		args       args
+		wantString string
+	}{
+		{
+			name:       "Capitals",
+			args:       args{key: "KEY"},
+			wantString: "key",
+		},
+		{
+			name:       "Capitals with underscore",
+			args:       args{key: "KEY_FOO"},
+			wantString: "key.foo",
+		},
+		{
+			name:       "Capitals with double underscore",
+			args:       args{key: "KEY__UNDERSCORE"},
+			wantString: "key_underscore",
+		},
+		{
+			name:       "Capitals with double and single underscore",
+			args:       args{key: "KEY_WITH__UNDERSCORE_AND__MORE"},
+			wantString: "key.with_underscore.and_more",
+		},
+		{
+			name:       "Capitals with triple underscore",
+			args:       args{key: "KEY___DASH"},
+			wantString: "key-dash",
+		},
+		{
+			name:       "capitals with double,triple and single underscore",
+			args:       args{key: "KEY_WITH___DASH_AND___MORE__UNDERSCORE"},
+			wantString: "key.with-dash.and-more_underscore",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if result := ConvertKey(tt.args.key); result != tt.wantString {
+				t.Errorf("ConvertKey() result = %v, wantStr %v", result, tt.wantString)
+			}
+		})
+	}
+}
+
+func Test_buildProperties(t *testing.T) {
+	type args struct {
+		spec        ConfigSpec
+		environment map[string]string
+	}
+	tests := []struct {
+		name string
+		args args
+		want map[string]string
+	}{
+		{
+			name: "only defaults",
+			args: args{
+				spec: ConfigSpec{
+					Prefixes: map[string]bool{},
+					Excludes: []string{},
+					Renamed:  map[string]string{},
```

Comment:
nit: no need to specify default values. default map is nil. Initialize here only if you specifically want it to be empty initialized map.